examples/oneapi/dpc/source/decision_forest/df_reg_traverse_model.cpp

```diff
@@ -0,0 +1,95 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#define ONEDAL_DATA_PARALLEL
+#include "oneapi/dal/algo/decision_forest.hpp"
+#include "oneapi/dal/io/csv.hpp"
+
+#include "example_util/utils.hpp"
+#include "oneapi/dal/exceptions.hpp"
+
+namespace dal = oneapi::dal;
+namespace df = dal::decision_forest;
+
+/** Visitor class, prints out tree nodes of the model when it is called back by model traversal method */
+template <typename Task>
+struct print_node_visitor {
+    bool operator()(const df::leaf_node_info<Task> &info) {
+        for (std::int64_t i = 0; i < info.get_level(); ++i)
+            std::cout << "  ";
+        std::cout << "Level " << info.get_level()
+                  << ", leaf node. Response value = " << info.get_label()
+                  << ", Impurity = " << info.get_impurity()
+                  << ", Number of samples = " << info.get_sample_count() << std::endl;
+        return true;
+    }
+
+    bool operator()(const df::split_node_info<Task> &info) {
+        for (std::int64_t i = 0; i < info.get_level(); ++i)
+            std::cout << "  ";
+        std::cout << "Level " << info.get_level()
+                  << ", split node. Feature index = " << info.get_feature_index()
+                  << ", feature value = " << info.get_feature_value()
+                  << ", Impurity = " << info.get_impurity()
+                  << ", Number of samples = " << info.get_sample_count() << std::endl;
+        return true;
+    }
+};
+
+template <typename Task>
+void print_model(const df::model<Task> &m) {
+    std::cout << "Number of trees: " << m.get_tree_count() << std::endl;
+    for (std::int64_t i = 0, n = m.get_tree_count(); i < n; ++i) {
+        std::cout << "Tree #" << i << std::endl;
+        m.traverse_dfs(i, print_node_visitor<df::task::regression>{});
+    }
+}
+
+void run(sycl::queue &q) {
```

Comment:
Something happened to formatting
```suggestion
void run(sycl::queue& q) {
```

Reply:
fixed